src/components/ec-table-filter/ec-table-filter.vue

```diff
@@ -0,0 +1,167 @@
+    class="ec-table-filter"
+    data-test="ec-table-filter"
+  >
+    <component
+      :is="filter.component"
+      v-for="(filter, index) in filters"
+      :key="index"
+      :value="value[filter.name]"
+      v-bind="filter"
+      :label="filter.label"
+      :data-test="`ec-table-filter__filter-item-${index}`"
+      class="ec-table-filter__filter-item"
+      @change="onChange(filter.name, $event)"
+    />
+    <!-- TODO ONL-4893
+    <button
+      type="button"
+      data-test="ec-table-filter__less-filters-button"
+      class="ec-table-filter__less-filters-button"
+      @click="handleLessFilterClick()"
+    >
+      {{ setLessFiltersButtonText }}
+      <ec-icon
+        class="ec-table-filter__less-filters-button--icon"
+        :size="16"
+        :name="setLessFiltersButtonIcon"
+        type="interactive"
+      />
+    </button>
+    -->
+    <button
+      v-if="hasFilters"
+      type="button"
+      data-test="ec-table-filter__clear-filters-button"
+      class="ec-table-filter__clear-filters-button"
+      @click="clearFilters()"
+    >
+      {{ clearFiltersButtonText }}
+    </button>
+  </section>
+</template>
+<script>
+import EcIcon from '../ec-icon';
+
+export default {
+  name: 'EcTableFilter',
+  components: { EcIcon },
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    value: {
+      type: Object,
+      default: () => ({}),
+      required: false,
+    },
+    filters: {
+      type: Array,
+      required: true,
+      default: () => ([]),
+    },
+    lessFiltersButtonText: {
+      type: Array,
+      required: false,
+      default: () => (['Less filters', 'More filters']),
+    },
+    clearFiltersButtonText: {
+      type: String,
+      required: false,
+      default: 'Clear filters',
+    },
+    popoverOptions: {
+      type: Object,
+    },
+  },
+  data() {
+    return {
+      isLessFilterActive: true,
+    };
+  },
+  computed: {
+    hasFilters() {
+      return !!Object.keys(this.value).length;
+    },
+    // TODO ONL-4893
+    // setLessFiltersButtonIcon() {
+    //   return this.isLessFilterActive ? 'simple-chevron-down' : 'simple-chevron-up';
+    // },
+    // setLessFiltersButtonText() {
+    //   return this.isLessFilterActive ? this.lessFiltersButtonText[1] : this.lessFiltersButtonText[0];
+    // },
+  },
+  methods: {
+    // TODO ONL-4893
+    // handleLessFilterClick() {
+    //   this.isLessFilterActive = !this.isLessFilterActive;
+    // },
+    onChange(filterName, value) {
+      if (!value || value.length === 0) {
+        const { [filterName]: currentValue, ...otherFilters } = this.value;
+
+        this.$emit('change', otherFilters);
+      } else {
+        this.$emit('change', { ...this.value, [filterName]: value });
+      }
+    },
+    clearFilters() {
+      this.$emit('change', {});
+    },
+  },
+};
+
+</script>
+
+<style>
+.ec-table-filter {
+  @apply tw-block;
+  @apply tw-bg-gray-7;
+
+  @media screen and (min-width: theme('screens.sm')) {
```

Comment:
With tailwind, we do mobile first. so everything should be styled for smallest screen and then we should be using `@screen md {...}` for overriding it.

Reply:
Here I'm targetting the xs screens first. sm up should have white bg